### ISSUE TITLE
The answer result is confusing when calling multiple Lora models use like jmeter 

### ISSUE THREAD

**Issue by WangxuP** [huggingface/peft#1378]
#System Info

python -V
Python 3.10.11

python -m pip list
Package                       Version
-----------       ------------
accelerate                    0.23.0
aiofiles                      23.2.1
aiohttp                       3.9.1
aiosignal                     1.3.1
annotated-types               0.6.0
async-timeout                 4.0.3
attrs                         23.2.0
certifi                       2023.7.22
charset-normalizer            3.3.0
cmake                         3.27.7
datasets                      2.16.1
deepspeed                     0.12.6
dill                          0.3.7
einops                        0.7.0
filelock                      3.12.4
frozenlist                    1.4.1
fsspec                        2023.9.2
h11                           0.9.0
hjson                         3.1.0
httpcore                      0.11.1
httptools                     0.6.1
httpx                         0.15.4
huggingface-hub               0.20.2
idna                          3.4
Jinja2                        3.1.2
lit                           17.0.3
MarkupSafe                    2.1.3
mpmath                        1.3.0
multidict                     5.2.0
multiprocess                  0.70.15
networkx                      3.2
ninja                         1.11.1.1
numpy                         1.26.1
nvidia-cublas-cu11            11.10.3.66
nvidia-cuda-cupti-cu11        11.7.101
nvidia-cuda-nvrtc-cu11        11.7.99
nvidia-cuda-runtime-cu11      11.7.99
nvidia-cudnn-cu11             8.5.0.96
nvidia-cufft-cu11             10.9.0.58
nvidia-curand-cu11            10.2.10.91
nvidia-cusolver-cu11          11.4.0.1
nvidia-cusparse-cu11          11.7.4.91
nvidia-nccl-cu11              2.14.3
nvidia-nvtx-cu11              11.7.91
packaging                     23.2
pandas                        2.1.4
peft                          0.5.0
pip                           23.3
psutil                        5.9.6
py-cpuinfo                    9.0.0
pyarrow                       14.0.2
pyarrow-hotfix                0.6
pydantic                      2.5.3
pydantic_core                 2.14.6
pynvml                        11.5.0
python-dateutil               2.8.2
pytz                          2023.3.post1
PyYAML                        6.0.1
regex                         2023.10.3
requests                      2.31.0
rfc3986                       1.5.0
safetensors                   0.4.0
sanic                         20.12.6
scipy                         1.11.3
setuptools                    68.0.0
six                           1.16.0
sniffio                       1.3.0
sympy                         1.12
tiktoken                      0.5.1
tokenizers                    0.13.3
torch                         2.0.1
tqdm                          4.66.1
transformers                  4.32.0
transformers-stream-generator 0.0.4
triton                        2.0.0
typing_extensions             4.8.0
tzdata                        2023.4
ujson                         5.8.0
urllib3                       2.0.7
uvloop                        0.18.0
websockets                    9.1
wheel                         0.41.2
xxhash                        3.4.1
yarl                          1.9.4

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

```
from peft import PeftModel
import torch
from transformers import AutoModelForCausalLM
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
model = AutoModelForCausalLM.from_pretrained(model_path, device_map="auto",
                                                              trust_remote_code=True, bf16=True).eval()

lora_model = PeftModel.from_pretrained(model, slot_lora_path, adapter_name="lora_1").cuda()
lora_model = PeftModel.from_pretrained(model, slot_lora_path, adapter_name="lora_2").cuda()
lora_model = PeftModel.from_pretrained(model, slot_lora_path, adapter_name="lora_3").cuda()
lora_model = lora_model.eval()

def func1():
    lora_model.set_adapter("lora_1")
    response, history = lora_model.chat(tokenizer,  prompt)
    return response

def func2():
    lora_model.set_adapter("lora_2")
    response, history = lora_model.chat(tokenizer,  prompt)
    return response

def func3():
    lora_model.set_adapter("lora_3")
    response, history = lora_model.chat(tokenizer,  prompt)
    return response

def main():
    # task_type is func1,... func3
    method_to_call = getattr(self, self.task_type, None)
    llm_answer = method_to_call(task_data)

from sanic import response
import asyncio
from sanic import Sanic
app = Sanic(name='big_model_server')

@app.route('/myapi', methods=['POST'])
async def qwen_14b_server(request):
    loop = asyncio.get_event_loop()
    resp= await loop.run_in_executor(None, main, request.json)
    return response.json(resp)

```
base model is qwen-14b(https://huggingface.co/Qwen/Qwen-14B-Chat)

**Error occurred when I calling multiple Lora models use jmeter  ?  help me !**

### Expected behavior

Output the inference results of each Lora model

**Comment by BenjaminBossan** [comment]
Could you please explain further what you expect to happen and what you actually observe? I assume that you want to use some load testing and that under some circumstances, the results are worse. Is that it?

**Comment by WangxuP** [comment]
> Could you please explain further what you expect to happen and what you actually observe? I assume that you want to use some load testing and that under some circumstances, the results are worse. Is that it?

My expected outcome is that each Lora small model can accomplish its own tasks, such as Lora1 being a Chinese English translation and Lora2 being a translation between Chinese and French.
But what I have observed now is that if Lora1 is called continuously, sometimes Lora2 will have incorrect answers, such as garbled characters, etc. Of course, this is an occasional issue.

**Comment by BenjaminBossan** [comment]
Okay, so your question is not about load testing specifically. To make sure I understand correctly: If you probe Lora2 repeatedly, without Lora1 being involved, the results are fine. But when you probe Lora2 repeatedly while switching back and forth between Lora1 and Lora2, sometimes incorrect results are returned. Is that understanding correct?

I could imagine that this is a concurrency issue somehow. Do you send multiple requests concurrently via jmeter? Does the error still occur when you only send one concurrent request?

**Comment by WangxuP** [comment]
> Okay, so your question is not about load testing specifically. To make sure I understand correctly: If you probe Lora2 repeatedly, without Lora1 being involved, the results are fine. But when you probe Lora2 repeatedly while switching back and forth between Lora1 and Lora2, sometimes incorrect results are returned. Is that understanding correct?
> 
> I could imagine that this is a concurrency issue somehow. Do you send multiple requests concurrently via jmeter? Does the error still occur when you only send one concurrent request?

In fact, I did not actually use jmeter for concurrent requests, and when accessing only one Lora, I did not find any confusion in the results.
I used jmeter as an example just to illustrate that I do frequently call Lora1 and Lora2 during the actual trial process.
When I repeatedly detect Lora1 and Lora2 after starting them, sometimes incorrect results are returned.

**Comment by BenjaminBossan** [comment]
I see. Do you have a script that you could share to reproduce the issue? As I mentioned, it could maybe be a concurrency issue, as switching adapters is not atomic, but it's hard to say without a way to reproduce the error.
